server/data/baseClientApiClient.test.ts

```diff
@@ -0,0 +1,208 @@
+import nock from 'nock'
+import config from '../config'
+import {
+  clientSecretsResponseFactory,
+  getBaseClientResponseFactory,
+  listBaseClientResponseFactory,
+  listClientInstancesResponseFactory,
+  updateBaseClientDeploymentFactory,
+} from '../testutils/factories'
+import BaseClientApiClient from './baseClientApiClient'
+import addBaseClientRequest from '../testutils/factories/requests/addBaseClientRequest'
+import updateBaseClientRequest from '../testutils/factories/requests/updateBaseClientRequest'
+
+jest.mock('./tokenStore')
+
+const token = { access_token: 'token-1', expires_in: 300 }
+
+describe('baseClientApiClient', () => {
+  let baseClientApi: nock.Scope
+  let baseClientApiClient: BaseClientApiClient
+
+  beforeEach(() => {
+    baseClientApi = nock(config.apis.hmppsAuthorizationServer.url)
+    baseClientApiClient = new BaseClientApiClient(token.access_token)
+  })
+
+  afterEach(() => {
+    jest.resetAllMocks()
+    nock.cleanAll()
+  })
+
+  const mockSuccessfulBaseClientRestApiCall = <TReturnData>(url: string, returnData: TReturnData) => {
+    baseClientApi.get(url).matchHeader('authorization', `Bearer ${token.access_token}`).reply(200, returnData)
+  }
+
+  const mockSuccessfulBaseClientRestApiPostCall = <TReturnData>(url: string, returnData: TReturnData) => {
+    baseClientApi.post(url).matchHeader('authorization', `Bearer ${token.access_token}`).reply(200, returnData)
+  }
+
+  const mockBaseClientRestApiPutCall = <TReturnData>(url: string, status: number, returnData: TReturnData) => {
+    baseClientApi.put(url).matchHeader('authorization', `Bearer ${token.access_token}`).reply(status, returnData)
+  }
+  const mockBaseClientRestApiDeleteCall = <TReturnData>(url: string, status: number, returnData: TReturnData) => {
+    baseClientApi.delete(url).matchHeader('authorization', `Bearer ${token.access_token}`).reply(status, returnData)
+  }
+
+  describe('listBaseClients', () => {
+    it('Should return data from the API', async () => {
+      // Given the network is mocked to return a response
+      const testResponse = listBaseClientResponseFactory.build()
+      mockSuccessfulBaseClientRestApiCall(`/base-clients`, testResponse)
+
+      // When we call the API client
+      const promise = baseClientApiClient.listBaseClients()
```

Comment:
Make one line? Just as simple?

`const output = await baseClientApiClient.listBaseClients()`

Reply:
yep - extracted to two lines to aid debugging - will replace